odml/property.py

```diff
@@ -96,7 +97,23 @@ def dtype(self, new_type):
     @property
     def parent(self):
         """the section containing this property"""
-        return self._section
+        return self._parent
+
+    @parent.setter
+    def parent(self, new_parent):
+        if new_parent is None:
+            return
+        elif self._validate_parent(new_parent):
+            self._parent = new_parent
+            self._parent.append(self)
+        else:
+            raise ValueError("odml.Property.parent: passed value is not of consistent type! odml.Section expected")
+
+    def _validate_parent(self, new_parent):
+        from odml.section import BaseSection
```

Comment:
Since we check for instance of Property in section.py, it does create a cycle here.